src/frontend/js/components/DashboardCreditCardsManagement/index.spec.tsx

```diff
@@ -0,0 +1,437 @@
+  });
+
+  it('renders the correct label for an expiration date that will not soon expire', async () => {
+    const limit = new Date();
+    limit.setMonth(limit.getMonth() + 4);
+    const date = faker.date.future(0.2, limit);
+    const creditCard: CreditCard = {
+      ...mockFactories.CreditCardFactory.generate(),
+      expiration_month: date.getMonth() + 1,
+      expiration_year: date.getFullYear(),
+    };
+    fetchMock.get('https://joanie.endpoint/api/credit-cards/', [creditCard]);
+    const client = createQueryClientWithUser(true);
+    await act(async () => {
+      render(
+        <QueryClientProvider client={client}>
+          <IntlProvider locale="en">
+            <SessionProvider>
+              <DashboardTest initialRoute={DashboardPaths.PREFERENCES} />
+            </SessionProvider>
+          </IntlProvider>
+        </QueryClientProvider>,
+      );
+    });
+    // No error is shown.
+    expect(screen.queryByText('An error occurred', { exact: false })).toBeNull();
+    const element = screen.getByText(
+      'Expires on ' +
+        (date.getMonth() + 1).toLocaleString(undefined, {
+          minimumIntegerDigits: 2,
+        }) +
+        '/' +
+        date.getFullYear(),
+    );
+    expect(element.className).toContain('dashboard-credit-card__expiration');
+    expect(element.className).not.toContain('dashboard-credit-card__expiration--expired');
+    expect(element.className).not.toContain('dashboard-credit-card__expiration--soon');
+  });
+
+  it('deletes a credit card', async () => {
+    const client = createQueryClientWithUser(true);
+    const creditCards = mockFactories.CreditCardFactory.generate(5);
+    fetchMock.get('https://joanie.endpoint/api/credit-cards/', creditCards);
+    await act(async () => {
+      render(
+        <QueryClientProvider client={client}>
+          <IntlProvider locale="en">
+            <SessionProvider>
+              <DashboardTest initialRoute={DashboardPaths.PREFERENCES} />
+            </SessionProvider>
+          </IntlProvider>
+        </QueryClientProvider>,
+      );
+    });
+    // No error is shown.
+    expect(screen.queryByText('An error occurred', { exact: false })).toBeNull();
+
+    // Find the delete button of the first credit card.
+    const creditCard = creditCards[0];
+    screen.getByText(creditCard.title);
+    const creditCardContainer = screen.getByTestId('dashboard-credit-card__' + creditCard.id);
+    const deleteButton = getByRole(creditCardContainer, 'button', {
+      name: 'Delete',
+    });
+
+    // Mock delete route and the refresh route to returns `creditCards` without the first one.
+    const deleteUrl = 'https://joanie.endpoint/api/credit-cards/' + creditCard.id + '/';
+    fetchMock.delete(deleteUrl, []);
+    fetchMock.get('https://joanie.endpoint/api/credit-cards/', creditCards.splice(1), {
+      overwriteRoutes: true,
+    });
+
+    // Clicking on the delete button calls delete API route.
```

Comment:
I'll add an assertion to ensure that credit card is `safeDelete`. As you mock the window.confirm, you are able to check if this method has been called.